gnocchi/rest/influxdb.py

```diff
@@ -0,0 +1,267 @@
+                 lambda s, loc, tok: tok[0][-1])))).setParseAction(
+                     lambda s, loc, tok: "".join(list(tok)))
+measurement = tag_key = tag_value = field_key = quoted_string | unquoted_string
+number = r"[+-]?\d+(:?\.\d*)?(:?[eE][+-]?\d+)?"
+number = pyparsing.Regex(number).setParseAction(
+    lambda s, loc, tok: float(tok[0]))
+integer = (
+    pyparsing.Word(pyparsing.nums).setParseAction(
+        lambda s, loc, tok: int(tok[0])) +
+    pyparsing.Suppress("i")
+    )
+field_value = integer | number | quoted_string
+timestamp = pyparsing.Word(pyparsing.nums).setParseAction(
+    lambda s, loc, tok: numpy.datetime64(int(tok[0]), 'ns'))
+
+line_protocol = (
+    measurement +
+    # Tags
+    pyparsing.Optional(pyparsing.Suppress(",") +
+                       pyparsing.delimitedList(
+                           pyparsing.OneOrMore(
+                               pyparsing.Group(
+                                   tag_key +
+                                   pyparsing.Suppress("=") +
+                                   tag_value), ",")).setParseAction(
+                                       lambda s, loc, tok: dict(list(tok))),
+                       default={}) +
+    pyparsing.Suppress(" ") +
+    # Fields
+    pyparsing.delimitedList(
+        pyparsing.OneOrMore(
+            pyparsing.Group(field_key +
+                            pyparsing.Suppress("=") +
+                            field_value), ",")).setParseAction(
+                                lambda s, loc, tok: dict(list(tok))) +
+    # Timestamp
+    pyparsing.Optional(pyparsing.Suppress(" ") + timestamp, default=None)
+).leaveWhitespace()
+
+
+query_parser = (
+    pyparsing.Suppress(pyparsing.CaselessLiteral("create")) +
+    pyparsing.Suppress(pyparsing.CaselessLiteral("database")) +
+    pyparsing.Suppress(pyparsing.White()) +
+    (pyparsing.QuotedString('"', escChar="\\") |
+     pyparsing.Word(pyparsing.alphas + "_",
+                    pyparsing.alphanums + "_")) +
+    pyparsing.Suppress(
+        pyparsing.Optional(pyparsing.Optional(pyparsing.White()) +
+                           pyparsing.Optional(pyparsing.Literal(";"))))
+)
+
+
+class InfluxDBController(rest.RestController):
+    _custom_actions = {
+        'ping': ['HEAD', 'GET'],
+        'query': ['POST'],
+        'write': ['POST'],
+    }
+
+    DEFAULT_TAG_RESOURCE_ID = "host"
+
+    @pecan.expose()
+    def ping(self):
+        pecan.response.headers['X-Influxdb-Version'] = (
+            "Gnocchi " + pbr.version.VersionInfo('gnocchi').version_string()
+        )
+
+    @pecan.expose('json')
+    def post_query(self, q=None):
+        if q is not None:
+            try:
+                query = query_parser.parseString(q)
+            except pyparsing.ParseException:
+                api.abort(501, {"cause": "Not implemented error",
+                                "detail": "q",
+                                "reason": "Querying is not implemented"})
```

Comment:
this error is confusing... i don't see how 'querying is not implemented', this is literally what the endpoint is. is it something 'unsupported' that will throw this?

Reply:
We don't support InfluxDB query language. I changed to "query not implemented" to be more precise about the fact that this query is not implemented.
The only supported is `CREATE DATABASE` that's not what I'd call a query ;)